cmd/monaco/download/download_configs.go

```diff
@@ -220,11 +226,17 @@ func downloadConfigs(downloaders downloaders, opts downloadConfigsOptions) (proj
 	}
 
 	if shouldDownloadAutomationResources(opts) {
-		automationCfgs, err := downloaders.Automation().Download(opts.projectName)
-		if err != nil {
-			return nil, err
+		if _, ok := downloaders.Automation().(automation.NoopAutomationDownloader); !ok {
```

Comment:
As far as I can see you have the Auth struct available in `opts` (though I'm not quite sure why/how clean that is tbh...)

To me it would seem less brittle to check for the existence of oAuth in the Auth struct, rather than against a specific implementation of the Automation downloader.

`if opts.auth.OAuth != nil {`